### PR TITLE
doc: snippets for admin clients with custom policies

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/1440)

### CODE REVIEW

File: google/cloud/spanner/samples/samples.cc [hunk 1]
```diff
@@ -2221,6 +2221,114 @@ void CustomRetryPolicy(std::vector<std::string> argv) {
...
+  }
+  //! [custom-instance-admin-policies]
+  namespace spanner = ::google::cloud::spanner;
+  [](std::string const& project_id) {
+    // An instance admin client is controlled by three policies. The retry
+    // policy determines for how long the client will retry transient failures.
+    auto retry_policy = spanner::LimitedTimeRetryPolicy(
+                            /*maximum_duration=*/std::chrono::minutes(25))
+                            .clone();
+    // The backoff policy controls how long does the client wait between
```
Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: s/wait/will wait/?

[Most of these are repeated in the DatabaseAdmin part.]

File: google/cloud/spanner/samples/samples.cc [hunk 2]
```diff
@@ -2221,6 +2221,114 @@ void CustomRetryPolicy(std::vector<std::string> argv) {
...
+    // retrying after a transient failure. Use an truncated exponential backoff
```
Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: s/an truncated/a truncated/

File: google/cloud/spanner/samples/samples.cc [hunk 3]
```diff
@@ -2221,6 +2221,114 @@ void CustomRetryPolicy(std::vector<std::string> argv) {
...
+    // with jitter to wait between retries:
+    //   https://en.wikipedia.org/wiki/Exponential_backoff
+    //   https://cloud.google.com/storage/docs/exponential-backoff
+    auto backoff_policy = spanner::ExponentialBackoffPolicy(
+                              /*initial_delay=*/std::chrono::seconds(2),
+                              /*maximum_delay=*/std::chrono::minutes(10),
+                              /*scaling=*/2.0)
+                              .clone();
+    // The polling policy controls how does the client wait for long running
```
Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: s/long running/long-running/

Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: s/how does the client wait/how the client waits/

File: google/cloud/spanner/samples/samples.cc [hunk 4]
```diff
@@ -2221,6 +2221,114 @@ void CustomRetryPolicy(std::vector<std::string> argv) {
...
+    // operations (such as creating new instances). `GenericPollingPolicy<>`
+    // combines existing policies.  In this case, wait for try for up to 45
```
Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: "wait for try for"?

File: google/cloud/spanner/samples/samples.cc [hunk 5]
```diff
@@ -2221,6 +2221,114 @@ void CustomRetryPolicy(std::vector<std::string> argv) {
...
+
+void CustomDatabaseAdminPolicies(std::vector<std::string> argv) {
+  if (argv.size() != 2) {
+    throw std::runtime_error(
+        "custom-database-admin-policies <project-id> <instance-id>");
+  }
+  //! [custom-database-admin-policies]
+  namespace spanner = ::google::cloud::spanner;
+  [](std::string const& project_id, std::string const& instance_id) {
+    // An database admin client is controlled by three policies. The retry
```
Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: /An database/A database/

File: google/cloud/spanner/samples/samples.cc (hunk 4 above)
Comment (tmatsuo) [inline google/cloud/spanner/samples/samples.cc]: /wait for try/keep trying/s
?

File: google/cloud/spanner/samples/samples.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/spanner/samples/samples.cc]: Fixed this one and the remaining comments.

**Comment by coryan** [comment]
PTAL.

File: google/cloud/spanner/samples/samples.cc [hunk 6]
```diff
@@ -2221,6 +2221,116 @@ void CustomRetryPolicy(std::vector<std::string> argv) {
...
+  }
+  //! [custom-instance-admin-policies]
+  namespace spanner = ::google::cloud::spanner;
+  [](std::string const& project_id) {
+    // An instance admin client is controlled by three policies. The retry
+    // policy determines for how long the client will retry transient failures.
+    auto retry_policy = spanner::LimitedTimeRetryPolicy(
+                            /*maximum_duration=*/std::chrono::minutes(25))
+                            .clone();
+    // The backoff policy controls how long does the client wait between
```
Comment (devbww) [inline google/cloud/spanner/samples/samples.cc]: "wait" to "waits" or "will wait" (and below).

Comment (coryan) [reply google/cloud/spanner/samples/samples.cc]: Thanks, that inspired me to change the paragraph even further. PTAL.

**Pull request merged**